pkg/rancher-desktop/config/settings.ts

```diff
@@ -230,9 +230,11 @@ export function load(deploymentProfiles: DeploymentProfileType): Settings {
       // because on Windows it's impossible to have an empty set of registry keys
       // but still have a deployment profile.
       // Linux and macOS can have empty files, but we'll ignore those.
-      if (Object.keys(deploymentProfiles.defaults).length) {
-        _.merge(settings, deploymentProfiles.defaults);
-        if (!_.has(deploymentProfiles.defaults, 'virtualMachine.memoryInGB')) {
+      if (Object.keys(deploymentProfiles.defaults).length || Object.keys(deploymentProfiles.locked).length) {
+        if (Object.keys(deploymentProfiles.defaults).length) {
+          _.merge(settings, deploymentProfiles.defaults);
+        }
+        if (!_.has(settings, 'virtualMachine.memoryInGB') && !_.has(deploymentProfiles.locked, 'virtualMachine.memoryInGB')) {
           setDefaultMemory = true;
         }
       } else {
```

Comment:
I don't understand why the change is needed, and I don't understand why you need to check for `keys` at all: isn't the merge a no-op when you merge in an empty object?
And how does having a locked profile affect the merging at all?

Wouldn't do this perform the same action:

```suggestion
      _.merge(settings, deploymentProfiles.defaults);
      if (!_.has(settings, 'virtualMachine.memoryInGB') && !_.has(deploymentProfiles.locked, 'virtualMachine.memoryInGB')) {
        setDefaultMemory = true;
      } else {
```

Reply:
Yes, that's correct. I thought of this biking home, and arrived at the same code.